### DISCUSSION TITLE
Should glb json begin with buffer field ?

### DISCUSSION THREAD

**Discussion by jimver04** [zeux/meshoptimizer#817] [Q&A] [answered]
Hi, 
I am experimenting on changes in the meshopt, so as to export without indices coding and to change the order of items in json. 

For some reason when starting from buffer as first json item the three.js reader works: 
![image](https://github.com/user-attachments/assets/04f553ac-377f-411d-867e-65e4f4680d30)

whereas starting from asset and having buffer in the end as in the following, does not work. I suppose that it is a bug? 

![image](https://github.com/user-attachments/assets/6307ea7c-7f68-4c7b-bc51-1e0d011a7dc8)

Thanks,
Dimitrios Ververidis

**Comment by zeux** [comment]
Are you missing a comma before "buffers"?

Also, please use GitHub Discussions for things like this. Let me know if I can communicate this more clearly via issue templates, but this is not a bug in this project - it's something that only happens once you change the code of this project to be incorrect.

**Reply by zeux** [reply, marked as answer]
I've also adjusted the code to put the asset metadata & extensions first: https://github.com/zeux/meshoptimizer/pull/818
